### PR TITLE
[IMP] website_sale*: little improvements/cleanups

### PR DESCRIPTION
**Pull request by Feyensv**
Enterprise PR: https://github.com/odoo/enterprise/pull/45427

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/website_sale_picking/controllers/main.py [hunk 1]
```diff
@@ -2,27 +2,29 @@
...
 from odoo.addons.website_sale.controllers.payment import PaymentPortal
 
 
 class PaymentPortalOnsite(PaymentPortal):
 
-    def _validate_transaction_for_order(self, transaction, sale_order_id):
-        """
-        Throws a ValidationError if the user tries to pay on site without also using an onsite delivery carrier
+    def _validate_transaction_for_order(self, transaction, sale_order):
+        """Override to make sure the onsite provider is not used without the onsite carrier
```
Comment (AntoineVDV) [inline addons/website_sale_picking/controllers/main.py:12]:
```suggestion
        """ Override of `website_sale` to make sure the onsite provider is not used without the onsite carrier.
```

**Review by AntoineVDV — APPROVED** [review]
Nice work! :+1:

**Comment by AntoineVDV** [comment]
IMHO, the two first ci/security concerns (JSON routes and `csrf=False`) are valid, while I'm not too sure about what the third one (list expansion) is about; maybe check with runbot team? :thinking:

**Review by AntoineVDV — APPROVED** [review]
@robodoo r+ rebase-ff override=ci/security
Reason: List extension with only constants
Link to ci/security build (master): https://runbot.odoo.com/runbot/build/54402828

**Comment by Feyensv** [comment]
@robodoo r+

**Comment by Feyensv** [comment]
@robodoo r+

**Comment by Feyensv** [comment]
@robodoo r+

**Comment by Feyensv** [comment]
@robodoo r+

**Pull request closed without merging**